Assets/HoloToolkit/Utilities/Scripts/Solvers/SolverConstantViewSize.cs

```diff
@@ -0,0 +1,190 @@
+			{
+				objectSize = baseSize;
+			}
+			else
+			{
+				Debug.LogError("ConstantViewSize: Object base size calculate was 0, defaulting to 1");
+				objectSize = 1f;
+			}
+		}
+
+		/// <summary>
+		///   Attempts to calculate the size of the bounds which contains all child renderers.
+		///   This may be tricky to use, as this happens during initialization, while the app may
+		///   be undergoing scaling by other solvers/components.  Thus, the size calculation might
+		///   be inaccurate.  It's probably a better idea to use ManualObjectSize just to be sure.
+		/// </summary>
+		/// <returns> Object diameter </returns>
+		private float CalculateObjectSize()
+		{
+			if (ManualObjectSize > 0)
+			{
+				return ManualObjectSize;
+			}
+
+			Vector3 rootScale = transform.root.localScale;
+			transform.root.localScale = Vector3.one;
+
+			float maxSize = 1f;
+
+			Bounds combinedBounds = new Bounds(transform.position, Vector3.zero);
+			Renderer[] renderers = this.GetComponentsInChildren<Renderer>();
+
+			foreach (Renderer render in renderers)
+			{
+				combinedBounds.Encapsulate(render.bounds);
+			}
+
+			maxSize = combinedBounds.extents.magnitude;
+
+			transform.root.localScale = rootScale;
+
+			return maxSize;
+		}
+
+
+		public override void SolverUpdate()
+		{
+			float lastScalePct = objectScalePercent;
+			AdjustSizeForView(solverHandler.TransformTarget);
+			float scaleDiff = (objectScalePercent - lastScalePct) / solverHandler.DeltaTime;
+
+			if (scaleDiff > ScaleBuffer)
+			{
+				ScaleState = ScaleStateEnum.Growing;
+			}
+			else if (scaleDiff < -ScaleBuffer)
+			{
+				ScaleState = ScaleStateEnum.Shrinking;
+			}
+			else
+			{
+				ScaleState = ScaleStateEnum.Static;
+			}
+		}
+
+		/// <summary>
+		///   Returns the scale to be applied based on the FOV.  This scale will be multiplied by distance as part of
+		///   the final scale calculation, so this is the ratio of vertical fov to distance.
+		/// </summary>
+		/// <returns> Scale of vFOV </returns>
+		public float GetFOVScalar()
+		{
+            float camFOVrad = (CameraCache.Main.aspect * CameraCache.Main.fieldOfView) * Mathf.Deg2Rad;
+
+			float sinfov = Mathf.Sin(camFOVrad * 0.5f);
+			float scalar = 2f * TargetViewPercentV * sinfov / objectSize;
+
+			return scalar;
+		}
+
+		// Borrowed from HoloSkype?
+		private void AdjustSizeForView(Transform targTransform)
+		{
+			if (targTransform != null)
+			{
+				// Get current fov each time instead of trying to cache it.  Can never count on init order these days
+				fovScalar = GetFOVScalar();
+
+                // Set the linked alt scale ahead of our work.  This is an attempt to minimize jittering by having solvers work with an interpolated scale.
+                solverHandler.AltScale.SetGoal(this.transform.localScale);
+
+				// Calculate scale based on distance from view.  Do not interpolate so we can appear at a constant size if possible.  Borrowed from greybox.
+				//bool ignoreParent = false;
```

Comment:
So scale gets applied to parent transform?
How does this work if the target transform is the camera?